src/pal/src/exception/signal.cpp

```diff
@@ -494,6 +507,62 @@ static void sigquit_handler(int code, siginfo_t *siginfo, void *context)
     kill(gPID, code);
 }
 
+/*++
+ Function :
+    HandleExternalSignal
+
+    Write to a pipe to kick off handling of the signal.
+
+ Parameters :
+    signalCode - code of the external signal
+
+    (no return value)
+--*/
+static void HandleExternalSignal(int signalCode)
+{
+    BYTE signalCodeByte = (BYTE)signalCode;
+    ssize_t writtenBytes;
+    do
+    {
+        writtenBytes = write(g_signalPipe[1], &signalCodeByte, 1);
+    }
+    while ((writtenBytes == -1) && (errno == EINTR));
+
+    if (writtenBytes == -1)
+    {
+        // Fatal error
+        abort();
+    }
+}
+
+/*++
+Function :
+    sigterm_handler
+
+    handle SIGTERM signal
+
+Parameters :
+    POSIX signal handler parameter list ("man sigaction" for details)
+
+    (no return value)
+--*/
+static void sigterm_handler(int code, siginfo_t *siginfo, void *context)
+{
+    if (PALIsInitialized())
+    {
+        HandleExternalSignal(code);
+    }
+    else
+    {
+        TRACE("SIGTERM signal was unhandled; chaining to previous sigaction\n");
```

Comment:
I am going to remove all the TRACE calls from signal handlers soon, since I have found they invoke functions that are not signal safe (e.g. string formatting). So I would suggest not adding it here now.